src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -44,6 +205,469 @@ private void BreakPointWithDebuggerFuncEval()
+
+                        var subscription = newListener.Subscribe(new CallbackObserver<KeyValuePair<string, object>>(delegate (KeyValuePair<string, object> evnt)
+                        {
+                            // THe filter is only a best effort here we do it for real.   
+                            if (eventNameFilter != null && eventNameFilter != evnt.Key)
+                                return;
+
+                            var outputArgs = this.Morph(evnt.Value);
+#if !NO_EVENTSOURCE_COMPLEX_TYPE_SUPPORT
+                            _eventSource.Event(newListener.Name, evnt.Key, outputArgs);
+#else
+                            _eventSource.EventJSON(newListener.Name, evnt.Key, ToJSON(outputArgs));
+#endif
+                        }), eventNameFilterPredicate);
+                        _liveSubscriptions = new Subscriptions(subscription, _liveSubscriptions);
+                    }
+                }));
+            }
+
+            private void Dispose()
+            {
+                if (_diagnosticsListenersSubscription != null)
+                {
+                    _diagnosticsListenersSubscription.Dispose();
+                    _diagnosticsListenersSubscription = null;
+                }
+
+                if (_liveSubscriptions != null)
+                {
+                    var subscr = _liveSubscriptions;
+                    _liveSubscriptions = null;
+                    while (subscr != null)
+                    {
+                        subscr.Subscription.Dispose();
+                        subscr = subscr.Next;
+                    }
+                }
+            }
+
+            public List<KeyValuePair<string, string>> Morph(object args)
+            {
+                // Transform the args into a bag of key-value strings.  
+                var outputArgs = new List<KeyValuePair<string, string>>();
+                if (args != null)
+                {
+                    if (!_noImplicitTransforms)
+                    {
+                        Type argType = args.GetType();
+                        if (_expectedArgType != argType)
+                        {
+                            // Figure out the default properties to send on to EventSource.  These are all string or primitive properties.  
+                            _implicitTransforms = null;
+                            TransformSpec newSerializbleArgs = null;
+                            TypeInfo curTypeInfo = argType.GetTypeInfo();
+                            foreach (var property in curTypeInfo.DeclaredProperties)
+                            {
+                                var propertyType = property.PropertyType;
+                                if (propertyType == typeof(string) || propertyType.GetTypeInfo().IsPrimitive)
+                                    newSerializbleArgs = new TransformSpec(property.Name, 0, property.Name.Length, newSerializbleArgs);
```

Comment:
If it's some other type, we just ignore the property silently?  Just making sure that's the desired behavior.